test/unit/abilities/provider_any_test.rb

```diff
@@ -69,6 +69,47 @@ def test_policies_not_tenant
       assert_cannot ability, :show, Accounts::AccountCreatedEvent.create(account, user)
     end
 
+    class ShowAlertRelatedEventTest < ProviderAnyTest
+      setup do
+        user.stubs(:has_permission?)
+
+        cinstance = FactoryBot.build_stubbed(:simple_cinstance)
+        alert = FactoryBot.build_stubbed(:limit_violation, cinstance: cinstance)
+        @limit_violation_reached_provider_event = Alerts::LimitViolationReachedProviderEvent.create(alert)
+      end
+
+      attr_reader :limit_violation_reached_provider_event
+
+      test 'cannot show AlertRelatedEvent when user does not have :monitoring' do
+        user.expects(:has_permission?).with(:monitoring).returns(false)
+
+        assert_cannot ability, :show, limit_violation_reached_provider_event
+      end
+
+      test 'can show AlertRelatedEvent when user has :monitoring and does not have a service' do
+        user.expects(:has_permission?).with(:monitoring).returns(true)
+
+        assert_can ability, :show, limit_violation_reached_provider_event
+      end
```

Comment:
I don't understand this test. The user can receive alerts for all services when it doesn't have permission over any of them?

Reply:
Oh... That's a bad name. Should be something like "can show AlertRelatedEvent when user has :monitoring and the alert is not related to a specific service".
Updated in [ea018de](https://github.com/3scale/porta/pull/3134/commits/ea018dee8c8642478003730a4bfb9ffe5f48cd97).

Reply:
Oh... That's a bad name.
Should be something like "can show AlertRelatedEvent when user has :monitoring and the user has no services (old permissions)".
Which's true because of [this comment](https://github.com/3scale/porta/blob/master/app/models/user/permissions.rb#L96-L98). Basically, we're supporting an old permission system where users have no services.
Updated in [ea018de](https://github.com/3scale/porta/pull/3134/commits/ea018dee8c8642478003730a4bfb9ffe5f48cd97).